streams/src/test/java/org/apache/kafka/streams/integration/AdjustStreamThreadCountTest.java

```diff
@@ -421,6 +421,7 @@ public void shouldResizeMaxBufferAfterThreadRemovalTimesOut() throws Interrupted
     }
 
     @Test
+    @SuppressWarnings("deprecation")
```

Comment:
Many of the tests in this PR are just testing deprecated methods, so it's good to just suppress them. But this one is using a deprecated method to test something else. I think we should try to minimize add-on changes, since this PR is already so big, but can you add a TODO to follow up in a subsequent PR to migrate this and tests like it (similar to what we did for the new PAPI)?